src/domain/headlessCms/persistor.ts

```diff
@@ -0,0 +1,203 @@
+  /**
+   * The logging level to use. Defaults to LogLevel.INFO.
+   */
+  logLevel?: LogLevel;
+}
+
+export class TimedApolloCachePersistor {
+  persistorLocalStorageKey: string;
+  timePersistedLocalStorageKey: string;
+  persistedCacheTimeToLive: number;
+  persistor: CachePersistor<NormalizedCacheObject>;
+  private logger: TimedApolloCachePersistorLogger;
+
+  /**
+   * Creates a new TimedApolloCachePersistor instance to manage the Apollo cache
+   * with an expiration time. This class automatically clears expired caches
+   * and provides methods for persisting and restoring the cache to/from localStorage.
+   *
+   * @param cache The Apollo InMemoryCache instance to be managed.
+   * @param options Options for configuring the persistor.
+   *
+   * @example
+   * const cache = new InMemoryCache();
+   * const persistor = new TimedApolloCachePersistor(cache, {
+   * persistedCacheTimeToLive: 1000 * 60 * 5, // 5 minute expiry
+   * logLevel: LogLevel.DEBUG
+   * });
+   */
+  constructor(
+    cache: InMemoryCache,
+    options: TimedApolloCachePersistorOptions = {}
+  ) {
+    const {
+      persistedCacheTimeToLive = 1000 * 60 * 10, // 10 minutes
+      logLevel = LogLevel.INFO,
+    } = options;
+    this.logger = new TimedApolloCachePersistorLogger(logLevel);
+    this.persistorLocalStorageKey = `kukkuu-ui-cms-${AppConfig.environment}-apollo-cache`;
+    this.timePersistedLocalStorageKey = `kukkuu-ui-cms-${AppConfig.environment}-apollo-cache-persisted-at`;
+    this.persistedCacheTimeToLive = persistedCacheTimeToLive;
+    this.persistor = this.createApolloCachePersistor(cache);
+  }
+
+  /**
+   * @private
+   * Creates and configures the Apollo CachePersistor instance.
+   * This method also handles clearing expired caches from localStorage.
+   * * @param cache The Apollo InMemoryCache instance.
+   * @returns The configured CachePersistor instance.
+   */
+  private createApolloCachePersistor(cache: InMemoryCache) {
+    if (this.hasPersistedCacheExpired()) {
+      try {
+        this.clearLocalStorage();
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } catch (e: any) {
+        this.logger.error('Failed to clear local storage:', e.message, e.stack);
+      }
+
+      localStorage.setItem(
+        this.timePersistedLocalStorageKey,
+        Date.now().toString()
+      );
+
+      this.logger.debug(
+        'timePersistedLocalStorageKey has been set to',
+        this.timePersistedLocalStorageKey
+      );
+    }
+    return new CachePersistor({
+      cache,
+      storage: new LocalStorageWrapper(localStorage),
+      key: this.persistorLocalStorageKey,
+    });
+  }
+
+  /**
+   * Checks if the persisted Apollo cache has expired based on the
+   * `persistedCacheTimeToLive` value.
+   *
+   * @returns `true` if the cache has expired, `false` otherwise.
+   */
+  hasPersistedCacheExpired() {
+    const persistedAt = parseInt(
+      localStorage.getItem(this.timePersistedLocalStorageKey) || ''
```

Comment:
```suggestion
      localStorage.getItem(this.timePersistedLocalStorageKey) ?? ''
```
Unless there's a specific reason for wanting to handle other falsy values beside null and undefined also here? [MSDN](https://developer.mozilla.org/en-US/docs/Web/API/Storage/getItem) says getItem's return value is `A string containing the value of the key. If the key does not exist, null is returned.` so `??` will handle the "not found" case. Also SonarCloud will probably raise a new issue related to https://rules.sonarsource.com/typescript/RSPEC-6606/ i.e. "Nullish coalescing should be preferred".